source/common/grpc/google_async_client_impl.cc

```diff
@@ -235,6 +235,13 @@ void GoogleAsyncStreamImpl::closeStream() {
 
 void GoogleAsyncStreamImpl::resetStream() {
   ENVOY_LOG(debug, "resetStream");
+  // The gRPC API requires calling Finish() at the end of a stream, even
```

Comment:
It would be ideal to have some unit test to verify the contract on the gRPC operations is valid. This woudl involve having a mock and dependency injection for `rw_`. Do you know if there is something tractable here?

Reply:
That may be possible, but I suspect it would be a lot of work.

As an alternative, are there tests for this code that exercise `resetStream()` and run under ASAN?  Any such test would presumably fail once the gRPC change goes in but without this PR.  We could verify that to make sure we have coverage for this.

Reply:
All tests run under ASAN and we have a bunch of tests in `test/common/grpc/grpc_client_integration_test.cc` which should exercise this path. Could you check to see if they provide the required coverage?

Reply:
OK, let's ship it, I think this is fine. It's definitely in the exception bucket, but the difficulty in writing tests here seems to outweigh the real improvement we get from respecting the contract.